commandapi-core/src/main/java/dev/jorel/commandapi/Tooltip.java

```diff
@@ -90,7 +257,48 @@ public static <S> Tooltip<S>[] arrayOf(Tooltip<S>... tooltips) {
 	 * @return the mapping function from this tooltip into a StringTooltip
 	 */
 	public static <S> Function<Tooltip<S>, StringTooltip> build(Function<S, String> mapper) {
-		return t -> StringTooltip.of(mapper.apply(t.object), t.tooltip);
+		return t -> StringTooltip.ofMessage(mapper.apply(t.object), t.tooltip);
 	}
-	
+
+	/**
+	 * Converts an unformatted string to an unformatted tooltip by wrapping as with a {@link LiteralMessage}.
+	 *
+	 * If formatting is required, please see {@link #messageFromBaseComponents(BaseComponent...)},
+	 * or consider using the more modern adventure text api.
+	 *
+	 * @param string unformatted string tooltip
+	 * @return wrapped tooltip as a {@link LiteralMessage}
+	 */
+	public static Message messageFromString(String string) {
+		return new LiteralMessage(string);
+	}
+
+	/**
+	 * Converts a formatted bungee text component to a native minecraft text component which can be used natively by brigadier.
+	 *
+	 * This supports all forms of formatting including entity selectors, scores,
+	 * click &amp; hover events, translations, keybinds and more.
+	 *
+	 * Note: the bungee component api is deprecated, and the adventure text component api should be used instead
+	 *
+	 * @param components array of bungee text components
+	 * @return native minecraft message object which can be used natively by brigadier.
+	 */
+	public static Message messageFromBaseComponents(BaseComponent... components) {
+		return CommandAPIHandler.getInstance().getNMS().generateMessageFromJson(ComponentSerializer.toString(components));
+	}
+
+	/**
+	 * Converts a formatted adventure text component to a native minecraft text component which can be used natively by brigadier.
+	 *
+	 * This supports all forms of formatting including entity selectors, scores,
+	 * click &amp; hover events, translations, keybinds and more.
+	 **
+	 * @param component adventure text component
+	 * @return native minecraft message object which can be used natively by brigadier.
+	 */
+	public static Message messageFromAdventureComponent(Component component) {
+		return CommandAPIHandler.getInstance().getNMS().generateMessageFromJson(BukkitComponentSerializer.gson().serialize(component));
```

Comment:
`net.kyori.adventure.platform.bukkit.BukkitComponentSerializer` is not suitable for Paper Minecraft servers!
You should use the `net.kyori.adventure.text.serializer.gson.GsonComponentSerializer` instead.

The `BukkitComponentSerializer` is for CraftBukkit servers (non-Spigot and non-Paper) and is an external library that must be downloaded for CraftBukkit servers. `GsonComponentSerializer` comes packaged with Paper servers and that's the recommended class to use (as `PaperComponents.gsonSerializer()` [is deprecated](https://jd.papermc.io/paper/1.19/io/papermc/paper/text/PaperComponents.html#gsonSerializer()))